app/services/event_log/move_runner.rb

```diff
@@ -16,6 +16,7 @@ def call
         when Event::APPROVE
           move.status = Move::MOVE_STATUS_REQUESTED
           move.date = event.date
+          move.determine_supplier
```

Comment:
so once it is approved we need to determine again the supplier  ... yeah it sounds right, there might be other cases like this.
Also:
in case you change `determine_supplier`
this would be something like:
`move.supplier = determine_supplier`

Reply:
This has test coverage to ensure the supplier gets updated, so that would catch any implementation changes. I've covered all the cases where the move date can change too.